src/admm/environment.jl

```diff
@@ -164,16 +236,18 @@ mutable struct AdmmEnv{T,TD,TI,TM}
         env.params = Parameters()
         env.params.verbose = verbose
 
-        env.model = Model{T,TD,TI}(env.data, use_gpu, use_polar)
-        env.solution = Solution{T,TD}(env.data, env.model, rho_pq, rho_va)
-
-        wRIij = zeros(2*env.model.nline) # TODO: Not used
-
+        env.model = Model{T,TD,TI}(env.data, use_gpu, use_polar, use_twolevel)
         ybus = Ybus{Array{T}}(computeAdmitances(
             env.data.lines, env.data.buses, env.data.baseMVA; VI=Array{Int}, VD=Array{T})...)
 
-        init_solution!(env.solution, env.data, ybus, env.model.gen_start, env.model.line_start,
-                    rho_pq, rho_va, wRIij)
+        if !use_twolevel
+            env.solution = Solution{T,TD}(env.model)
+            init_solution!(env, ybus, rho_pq, rho_va)
```

Comment:
maybe we could use dispatch over `Solution` here? Something like
```julia
init_solution!(env, solution, ybus, rho_pq, rho_va)
```

Reply:
done